File: 8kyu 2/grow.js
/*
|--------------------------------------------------------------------------
| Beginner - Reduce but Grow
|--------------------------------------------------------------------------
|
*/

function grow(x) {
   return x.reduce((acc, curr) => acc * curr);
}


const grow = x => eval(x.join("*"))


function grow(x) {
   var product = 1;
   x.forEach(function (e) {
      product *= e
   })
   return product;
}


const grow = x => {
   let res = 1;
   for (let i = 0; i < x.length; i++) {
      res *= x[i];
   }
   return res;
};


console.log(grow([1, 2, 3]), 6);
console.log(grow([4, 1, 1, 1, 4]), 16);
console.log(grow([2, 2, 2, 2, 2, 2]), 64);

File: Python/8py 2/grow.py
"""
|--------------------------------------------------------------------------
| Beginner - Reduce but Grow
|--------------------------------------------------------------------------
|
"""

def grow(arr):
   res = 1
   for i in range(len(arr)):
      res *= arr[i]
   return res


from functools import reduce
def grow(arr):
   return reduce(lambda x, y: x * y, arr)


def grow(arr):
   return eval('*'.join([str(i) for i in arr]))


import math
def grow(arr):
   return math.prod(arr)


print(grow([1, 2, 3]), 6)
print(grow([4, 1, 1, 1, 4]), 16)
print(grow([2, 2, 2, 2, 2, 2]), 64)